sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -1276,6 +1280,14 @@ private async Task RunProcessingAsync(CancellationToken cancellationToken)
                 }))
                 .ConfigureAwait(false);
 
+            // If load balancing is greedy and there was a partition claimed, then signal that there should be no delay before
+            // invoking the next load balancing cycle.
+
+            if ((Options.LoadBalancingStrategy == LoadBalancingStrategy.Greedy) && (claimedOwnership != default))
+            {
```

Comment:
I am not sure if I am reading this correctly, but it looks like the delay is set to 0 only when the previous execution of the load balancer claimed a partition. Should the load balancer run with no delay if the current state is imbalanced (i.e. this instance should own more partitions) regardless of whether the claim happened or not? 

If a claim for a particular partition failed because another processor got to it first, this processor should not go back to waiting for `loadBalancingUpdateInterval` duration. Instead, it should run with no delay to attempt to claim another available partition. Ignore me if this is what is already happening.

Reply:
You are correct, and I overlooked that case.   I've updated the logic to flag that the load is believed to be balanced when:

- The processor owns at least its fair share of partitions.
- No partitions remain unclaimed
- The last load balancing cycle did not claim a partition (in case partitions are being stolen from an unresponsive processor)